src/action.ml

```diff
@@ -735,7 +735,16 @@ let rec exec t ~ectx ~dir ~env ~stdout_to ~stderr_to =
   | Run (Error e, _) ->
     Prog.Not_found.raise e
   | Run (Ok prog, args) ->
-    exec_run ~ectx ~dir ~env ~stdout_to ~stderr_to prog args
+    exec_run ~ectx ~dir ~env:(
+      match ectx.context with
```

Comment:
We shouldn't need this, `even` is initialized with `context.env` already

Reply:
Hmm yeah, I suppose we should also allow the user to override the ENV set here.